packages/core/src/test/amazonqFeatureDev/util/files.test.ts

```diff
@@ -28,5 +30,32 @@ describe('file utils', () => {
             assert.strictEqual(result.zipFileChecksum.length, 44)
             assert.strictEqual(telemetry.repositorySize, 24)
         })
+
+        // Test the logic that allows the customer to modify root source folder
+        it('throws a ToolkitError with ContentLengthError code', async function () {
+            const fileNamePrefix = 'file'
+            const testContent = 'test content'
+            let files = 0
+
+            let fileContent = ''
+            while (Buffer.byteLength(fileContent) < maxFileSizeBytes - testContent.length) {
+                fileContent += testContent
+            }
+
+            for (let accumulatedSize = 0; accumulatedSize <= maxRepoSizeBytes; files++) {
+                accumulatedSize += Buffer.byteLength(fileContent)
+            }
+
+            const workspace = await createTestWorkspace(files, { fileNamePrefix, fileContent })
+            const telemetry = new TelemetryHelper()
+
+            try {
+                await prepareRepoData([workspace.uri.fsPath], [workspace], telemetry, {
+                    record: () => {},
+                } as unknown as Metric<AmazonqCreateUpload>)
+            } catch (err) {
+                assert.strictEqual(err instanceof ContentLengthError, true)
```

Comment:
if `prepareRepoData` doesn't throw, the test will pass. is that intended? why not use `assert.throws` or `assert.rejects` (look for examples in the codebase)